src/app/routes/registration/enrolment-form/enrolment-form.component.spec.ts

```diff
@@ -304,6 +304,28 @@ describe('EnrolmentFormComponent', () => {
     const { submit } = getSelectors(hostDebug);
     expect(submit.nativeElement.disabled).toBeFalse();
   });
+
+  it('should emit values with empty strings', () => {
+    component.fieldList = [
+      {
+        fieldType: 'number',
+        label: 'label',
+        maxValue: 10,
+        minValue: 2,
+      },
+    ];
```

Comment:
Hmm, does it work if instead of sending in `{ key: 'label', value: undefined }`, you would just remove the `key/value` entirely (in this example, just send in an empty object)? I'm just a bit concerned that credential will be issued with properties that are empty strings, which I don't think is correct

Reply:
Sure, i can clear all key-value pairs that have undefined value when approving role.